src/main/java/com/intel/genomicsdb/GenomicsDBImporter.java

```diff
@@ -788,15 +701,17 @@ static File printLoaderJSONFile(
   /**
    * Creates a sorted list of callsets and generates unique TileDB
    * row indices for them. Sorted to maintain order between
-   * distributed share-nothing load processes.
+   * distributed share-nothing load processes. This method is synchronized
+   * to block multiple invocations (if by any chance) disturb the order in
+   * which TileDB row indexes are generated
    *
    * @param sampleNames list of sample names
    * @param useSamplesInOrderProvided  If True, do not sort the samples,
    *                                   use the order they appear in
    * @param lbRowIdx Smallest row idx which should be imported by this object
    * @return  Mappings of callset (sample) names to TileDB rows
    */
-  public static GenomicsDBCallsetsMapProto.CallsetMappingPB generateSortedCallSetMap(
+  public static synchronized GenomicsDBCallsetsMapProto.CallsetMappingPB generateSortedCallSetMap(
```

Comment:
nope, it is not updating shared structure. You're right -  synchronized only helps if two threads call the method from the 'same' object, not otherwise. I added this just to be safe.